File: tests/ut/cpp/ops/test_ops_maximum.cc
/**
 * Copyright 2020 Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
#include <vector>
#include <memory>
#include "common/common_test.h"
#include "ops/maximum.h"
#include "ir/dtype/type.h"
#include "abstract/dshape.h"
#include "utils/tensor_construct_utils.h"
namespace mindspore {
namespace ops {
class TestMaximum : public UT::Common {
 public:
  TestMaximum() {}
  void SetUp() {}
  void TearDown() {}
};

TEST_F(TestMaximum, test_ops_maximum) {
  auto maximum = std::make_shared<Maximum>();
  maximum->Init();
  auto tensor_x = TensorConstructUtils::CreateOnesTensor(kNumberTypeFloat32, std::vector<int64_t>{1, 3});
  auto tensor_y = TensorConstructUtils::CreateOnesTensor(kNumberTypeFloat32, std::vector<int64_t>{1, 3});
  MS_EXCEPTION_IF_NULL(tensor_x);
  MS_EXCEPTION_IF_NULL(tensor_y);
  auto maximum_abstract = maximum->Infer({tensor_x->ToAbstract(), tensor_y->ToAbstract()});
  MS_EXCEPTION_IF_NULL(maximum_abstract);
  EXPECT_EQ(maximum_abstract->isa<abstract::AbstractTensor>(), true);
  auto shape_ptr = maximum_abstract->BuildShape();
  MS_EXCEPTION_IF_NULL(shape_ptr);
  EXPECT_EQ(shape_ptr->isa<abstract::Shape>(), true);
  auto maximum_shape = shape_ptr->cast<abstract::ShapePtr>();
  MS_EXCEPTION_IF_NULL(maximum_shape);
  auto shape_vec = maximum_shape->shape();
  auto type = maximum_abstract->BuildType();
  MS_EXCEPTION_IF_NULL(type);
  EXPECT_EQ(type->isa<TensorType>(), true);
  auto tensor_type = type->cast<TensorTypePtr>();
  MS_EXCEPTION_IF_NULL(tensor_type);
  auto elem_type = tensor_type->element();
  EXPECT_EQ(elem_type->type_id(), kNumberTypeFloat32);
  EXPECT_EQ(shape_vec.size(), 2);
  EXPECT_EQ(shape_vec[0], 1);
  EXPECT_EQ(shape_vec[1], 3);
}

}  // namespace ops
}  // namespace mindspore

File: tests/st/ops/test_ops_maximum.py
# Copyright 2023 Huawei Technologies Co., Ltd
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# ============================================================================
from tests.mark_utils import arg_mark
import pytest
import numpy as np
from mindspore import context
from mindspore.ops import maximum

from tests.st.ops.test_ops_minimum import (minimum_maximum_case, minimum_maximum_case_vmap,
                                           minimum_maximum_case_all_dyn)

def np_maximum(input_x, input_y):
    return np.maximum(input_x, input_y)


@arg_mark(plat_marks=['platform_ascend', 'platform_gpu', 'cpu_linux', 'cpu_windows', 'cpu_macos'], level_mark='level1', card_mark='onecard', essential_mark='unessential')
@pytest.mark.parametrize('mode', [context.GRAPH_MODE, context.PYNATIVE_MODE])
def test_maximum(mode):
    """
    Feature: Test minimum op.
    Description: Test minimum.
    Expectation: the result match with expected result.
    """
    context.set_context(mode=mode)
    minimum_maximum_case(maximum, np_maximum, is_minimum=False)


@arg_mark(plat_marks=['platform_ascend', 'platform_gpu', 'cpu_linux', 'cpu_windows', 'cpu_macos'], level_mark='level1', card_mark='onecard', essential_mark='unessential')
@pytest.mark.parametrize('mode', [context.GRAPH_MODE, context.PYNATIVE_MODE])
def test_maximum_vmap(mode):
    """
    Feature: Test minimum op.
    Description: Test minimum vmap.
    Expectation: the result match with expected result.
    """
    context.set_context(mode=mode)
    minimum_maximum_case_vmap(maximum)


@arg_mark(plat_marks=['platform_ascend', 'platform_gpu', 'cpu_linux', 'cpu_windows', 'cpu_macos'], level_mark='level1', card_mark='onecard', essential_mark='unessential')
def test_maximum_all_dynamic():
    """
    Feature: Test minimum op.
    Description: Test minimum with both input and axis are dynamic.
    Expectation: the result match with expected result.
    """
    minimum_maximum_case_all_dyn(maximum)
